ITSystemTest failing when using standard JDK 17

Using standard JDK 17 to run tests is resulting in the following log output:

```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.01 s <<< FAILURE! - in com.google.cloud.oslogin.v1.it.ITSystemTest
[ERROR] com.google.cloud.oslogin.v1.it.ITSystemTest  Time elapsed: 0.004 s  <<< ERROR!
com.google.gson.JsonIOException: Failed making field 'java.time.Duration#seconds' accessible; either change its visibility or write a custom TypeAdapter for its declaring type
	at com.google.gson.internal.reflect.ReflectionHelper.makeAccessible(ReflectionHelper.java:22)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.getBoundFields(ReflectiveTypeAdapterFactory.java:158)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:101)
	at com.google.gson.Gson.getAdapter(Gson.java:501)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.createBoundField(ReflectiveTypeAdapterFactory.java:116)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.getBoundFields(ReflectiveTypeAdapterFactory.java:165)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:101)
	at com.google.gson.Gson.getAdapter(Gson.java:501)
	at com.google.gson.Gson.toJson(Gson.java:739)
	at com.google.gson.Gson.toJson(Gson.java:726)
	at com.google.gson.Gson.toJson(Gson.java:681)
	at com.google.gson.Gson.toJson(Gson.java:661)
	at com.google.cloud.oslogin.v1.it.ITSystemTest.getFromCredential(ITSystemTest.java:74)
	at com.google.cloud.oslogin.v1.it.ITSystemTest.setup(ITSystemTest.java:56)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.RunBefores.invokeMethod(RunBefores.java:33)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:107)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:83)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make field private final long java.time.Duration.seconds accessible: module java.base does not "opens java.time" to unnamed module @4d76f3f8
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
	at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:178)
	at java.base/java.lang.reflect.Field.setAccessible(Field.java:172)
	at com.google.gson.internal.reflect.ReflectionHelper.makeAccessible(ReflectionHelper.java:19)
	... 43 more
```

Environment:
1) Call `sdk use java 17.0.3-oracle`
2) `java -version`
```
java version "17.0.3" 2022-04-19 LTS
Java(TM) SE Runtime Environment (build 17.0.3+8-LTS-111)
Java HotSpot(TM) 64-Bit Server VM (build 17.0.3+8-LTS-111, mixed mode, sharing)
```
3) `mvn verify -Penable-integration-tests`

--- a/google-cloud-os-login/pom.xml
+++ b/google-cloud-os-login/pom.xml
@@ -121,8 +121,8 @@
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-surefire-plugin</artifactId>
         <configuration>
-          <argLine>--add-opens java.base/java.time=ALL-UNNAMED
-            --add-opens=java.base/java.math=ALL-UNNAMED</argLine>
+          <argLine>--add-opens=java.base/java.time=ALL-UNNAMED
+              --add-opens=java.base/java.math=ALL-UNNAMED</argLine>
         </configuration>
       </plugin>
         </plugins>

--- a/google-cloud-os-login/pom.xml
+++ b/google-cloud-os-login/pom.xml
@@ -125,7 +125,8 @@
             --add-opens=java.base/java.math=ALL-UNNAMED
             --add-opens=java.base/sun.security.rsa=ALL-UNNAMED
             --add-opens=java.base/sun.security.pkcs=ALL-UNNAMED
-            --add-opens=java.base/sun.security.x509=ALL-UNNAMED</argLine>
+            --add-opens=java.base/sun.security.x509=ALL-UNNAMED
+            --add-opens=java.base/sun.security.util=ALL-UNNAMED</argLine>
         </configuration>
       </plugin>
         </plugins>

--- a/google-cloud-os-login/pom.xml
+++ b/google-cloud-os-login/pom.xml
@@ -122,8 +122,8 @@
         <artifactId>maven-surefire-plugin</artifactId>
         <configuration>
           <argLine>--add-opens=java.base/java.time=ALL-UNNAMED
-              --add-opens=java.base/java.math=ALL-UNNAMED
-              --add-opens=java.base/sun.security.util=ALL-UNNAMED</argLine>
+            --add-opens=java.base/java.math=ALL-UNNAMED 
+            --add-opens=java.base/sun.security.rsa=ALL-UNNAMED</argLine>
         </configuration>
       </plugin>
         </plugins>

--- a/google-cloud-os-login/pom.xml
+++ b/google-cloud-os-login/pom.xml
@@ -122,8 +122,9 @@
         <artifactId>maven-surefire-plugin</artifactId>
         <configuration>
           <argLine>--add-opens=java.base/java.time=ALL-UNNAMED
-            --add-opens=java.base/java.math=ALL-UNNAMED 
-            --add-opens=java.base/sun.security.rsa=ALL-UNNAMED</argLine>
+            --add-opens=java.base/java.math=ALL-UNNAMED
+            --add-opens=java.base/sun.security.rsa=ALL-UNNAMED
+            --add-opens=java.base/sun.security.pkcs=ALL-UNNAMED</argLine>
         </configuration>
       </plugin>
         </plugins>

--- a/google-cloud-os-login/pom.xml
+++ b/google-cloud-os-login/pom.xml
@@ -121,7 +121,8 @@
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-surefire-plugin</artifactId>
         <configuration>
-          <argLine>--add-opens java.base/java.time=ALL-UNNAMED</argLine>
+          <argLine>--add-opens java.base/java.time=ALL-UNNAMED
+            --add-opens=java.base/java.math=ALL-UNNAMED</argLine>
         </configuration>
       </plugin>
         </plugins>

--- a/google-cloud-os-login/pom.xml
+++ b/google-cloud-os-login/pom.xml
@@ -124,7 +124,8 @@
           <argLine>--add-opens=java.base/java.time=ALL-UNNAMED
             --add-opens=java.base/java.math=ALL-UNNAMED
             --add-opens=java.base/sun.security.rsa=ALL-UNNAMED
-            --add-opens=java.base/sun.security=ALL-UNNAMED</argLine>
+            --add-opens=java.base/sun.security.pkcs=ALL-UNNAMED
+            --add-opens=java.base/sun.security.x509=ALL-UNNAMED</argLine>
         </configuration>
       </plugin>
         </plugins>

--- a/google-cloud-os-login/pom.xml
+++ b/google-cloud-os-login/pom.xml
@@ -126,7 +126,8 @@
             --add-opens=java.base/sun.security.rsa=ALL-UNNAMED
             --add-opens=java.base/sun.security.pkcs=ALL-UNNAMED
             --add-opens=java.base/sun.security.x509=ALL-UNNAMED
-            --add-opens=java.base/sun.security.util=ALL-UNNAMED</argLine>
+            --add-opens=java.base/sun.security.util=ALL-UNNAMED
+            --add-opens=java.base/java.security=ALL-UNNAMED</argLine>
         </configuration>
       </plugin>
         </plugins>

--- a/google-cloud-os-login/pom.xml
+++ b/google-cloud-os-login/pom.xml
@@ -124,7 +124,7 @@
           <argLine>--add-opens=java.base/java.time=ALL-UNNAMED
             --add-opens=java.base/java.math=ALL-UNNAMED
             --add-opens=java.base/sun.security.rsa=ALL-UNNAMED
-            --add-opens=java.base/sun.security.pkcs=ALL-UNNAMED</argLine>
+            --add-opens=java.base/sun.security=ALL-UNNAMED</argLine>
         </configuration>
       </plugin>
         </plugins>

--- a/google-cloud-os-login/pom.xml
+++ b/google-cloud-os-login/pom.xml
@@ -122,7 +122,8 @@
         <artifactId>maven-surefire-plugin</artifactId>
         <configuration>
           <argLine>--add-opens=java.base/java.time=ALL-UNNAMED
-              --add-opens=java.base/java.math=ALL-UNNAMED</argLine>
+              --add-opens=java.base/java.math=ALL-UNNAMED
+              --add-opens=java.base/sun.security.util=ALL-UNNAMED</argLine>
         </configuration>
       </plugin>
         </plugins>

--- a/google-cloud-os-login/pom.xml
+++ b/google-cloud-os-login/pom.xml
@@ -96,6 +96,7 @@
     </dependency>
   </dependencies>
 
+
   <profiles>
     <profile>
       <id>java9</id>
@@ -108,6 +109,24 @@
           <artifactId>javax.annotation-api</artifactId>
         </dependency>
       </dependencies>
+    </profile>
+    <profile>
+      <id>java17</id>
+      <activation>
+        <jdk>[17,)</jdk>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-surefire-plugin</artifactId>
+        <configuration>
+          <argLine>--add-opens java.base/java.time=ALL-UNNAMED</argLine>
+        </configuration>
+      </plugin>
+        </plugins>
+      </build>
+
     </profile>
   </profiles>
 

